api/include/opentelemetry/trace/default_span.h

```diff
@@ -63,7 +63,7 @@ class DefaultSpan : public Span
 
   nostd::string_view ToString() const noexcept { return "DefaultSpan"; }
 
-  DefaultSpan(SpanContext span_context) noexcept : span_context_(span_context) {}
+  DefaultSpan(const SpanContext &span_context) noexcept : span_context_(span_context) {}
```

Comment:
Not sure if this one is safe, due to:

```
opentracing-shim/src/tracer_shim.cc:        new opentelemetry::trace::DefaultSpan(span_context)};
```

Reply:
Reverted this change, just to be on the safe side. Instead, we move the copy into the member variable, similar to the latest change for [span_context.h](https://github.com/open-telemetry/opentelemetry-cpp/pull/3150#discussion_r1849343531)

Reply:
Thanks, the change is much better now.